### PR TITLE
Add CLINC150 Dataset

### PR DESCRIPTION
**Pull request by sumanthd17**
Added CLINC150 Dataset. The link to the dataset can be found [here](https://github.com/clinc/oos-eval) and the paper can be found [here](https://www.aclweb.org/anthology/D19-1131.pdf)

- [x] Followed the instructions in CONTRIBUTING.md
- [x] Ran the tests successfully
- [x] Created the dummy data

### CODE REVIEW

File: datasets/clinc_oos/clinc_oos.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        ClincConfig(
+            name=name, description=_DESCRIPTIONS[name], data_url=_DATA_URLS[name], citation=_CITATION, url=_URL
+        )
+        for name in ["small", "imbalanced", "plus"]
+    ]
+
+    def _info(self):
+        features = {}
+        features["text"] = datasets.Value("string")
+        features["intent"] = datasets.Value("string")
```
Comment (lhoestq) [inline datasets/clinc_oos/clinc_oos.py:97]: Can you change this to a `ClassLabel` feature type ?
for example
```
labels_list = ["oos", "translate", "transfer", "timer", "definition", ...]  # add here the 150 intents
features["intent"] = datasets.ClassLabel(names=labels_list)
```

**Review by lhoestq — COMMENTED** [review]
Awesome :) thanks for adding this one.

I added a comment suggesting to use the ClassLabel feature type for the intent column.

Once this in changed you can just re-generate the datasets_infos.json file using
```
datasets-cli test ./datasets/clinc_oos --save_infos --all_configs --ignore_verifications
```
and we'll be good to merge !

**Comment by lhoestq** [comment]
Looks like your PR now shows changes in many other files than the ones for CLINC150.
Feel free to create another branch and another PR

**Comment by sumanthd17** [comment]
create new [PR](https://github.com/huggingface/datasets/pull/1016)

closing this!

**Pull request closed without merging**